.circleci/config.yml

```diff
@@ -42,7 +42,7 @@ jobs:
           name: Danger
           when: always
           command: |
-            bundle exec danger
+            if [ -n "$DANGER_GITHUB_API_TOKEN" ]; then bundle exec danger; else echo "Skipping Danger for forked pull request."; fi
```

Comment:
@chrisballinger aren’t forked PRs where we want Danger the most, to enforce various things that community members may not be as used to as our internal team?

Reply:
@ZevEisenberg Yeah I'm not quite sure what to do about that, but it definitely would not be safe to turn on CircleCI's "share secrets with forks" feature. Previously we were skipping CI builds entirely for forks, which combined with our GitHub merge protection settings prevented us from merging any forked PRs. :)

At least if there are issues running the rest of the `danger` CI job, the job will still fail, and we'll have access to the build artifacts, test reports, and code coverage in CircleCI.

Reply:
Can we do anything with Danger without access to build secrets? Otherwise, maybe some kind of staging PR? Or can we run Danger out-of-band somehow?

Reply:
Unfortunately Danger can't post comments to GitHub without access to `$DANGER_GITHUB_API_TOKEN`, and I'm not comfortable with the scope of the token if it's leaked/exfiltrated when passed to a forked PR: 
* https://danger.systems/guides/getting_started.html#tokens-for-oss-projects
* https://danger.systems/guides/troubleshooting.html#i-want-to-only-run-danger-for-internal-branches

> We recommend giving the token the smallest scope possible. This means just `public_repo`, this scope is still ideally too much but this account shouldn’t have any access to other repos or organizations - so malicious use of the token is scoped to making new repos on it, or writing comments on other OSS projects. Because the token can be quite easily be extracted from the CI environment, this minimizes the chance for bad actors to cause chaos with it.

From GitHub's OAuth docs: 
> `public_repo` - Limits access to public repositories. That includes read/write access to code, commit statuses, repository projects, collaborators, and deployment statuses for public repositories and organizations. Also required for starring public repositories.

We could still run Danger "locally", which prints the results to the console, to still ensure that nothing has been broken in the Danger setup.

```
$ bundle exec danger pr $CIRCLE_PULL_REQUEST
...
Results:

Warnings:
- [ ] Big PR

Markdown:
## Current coverage for BonMot.framework is `82.51%`
✅ *No files affecting coverage found*
```

It would be interesting to set up a service that runs something similar to Danger via GitHub Action hooks, using the pre-baked `danger pr` markdown output from a CircleCI artifact. I know a few people out here are working on a hack project to make a Slack bot that spams you about your open PRs, perhaps we could use that as a template.